app/src/panels/label/panel-label.vue

```diff
@@ -1,16 +1,92 @@
 <script setup lang="ts">
-withDefaults(
+import { onMounted, onBeforeUnmount, onUpdated, ref } from 'vue';
+import { useAutoFontFit } from '@/composables/use-auto-fit-text';
+
+const props = withDefaults(
 	defineProps<{
 		showHeader?: boolean;
 		text?: string;
+		whiteSpace?: string | undefined;
 		color?: string | undefined;
+		textAlign?: string | undefined;
+		fontSize?: 'sm' | 'md' | 'lg' | 'auto';
+		fontWeight?: number | undefined;
+		fontStyle?: string | undefined;
+		font?: 'sans-serif' | 'serif' | 'monospace';
 	}>(),
 	{
 		showHeader: false,
 		text: '',
 		color: undefined,
+		whiteSpace: 'normal',
+		textAlign: 'center',
+		fontSize: 'auto',
+		fontWeight: 800,
+		fontStyle: undefined,
+		font: 'sans-serif',
 	}
 );
+
+const labelContainer = ref<HTMLDivElement | null>(null);
+const labelText = ref<HTMLParagraphElement | null>(null);
+
+const { adjustFontSize } = useAutoFontFit(labelContainer, labelText);
+
+let resizeObserver: ResizeObserver | null = null;
+
+function adjustPadding() {
+	const container = labelContainer.value;
+	if (!container) return '0px';
+
+	const paddingWidth = container.offsetWidth * 0.05;
+	const paddingHeight = container.offsetHeight * 0.05;
+
+	const padding = Math.round(Math.max(12, Math.min(paddingWidth, paddingHeight)));
+
+	if (props.showHeader == true) {
+		container.style.padding = '0px 12px 12px 12px';
+	} else {
+		container.style.padding = `${padding}px`;
+	}
+
+	return;
+}
+
+async function updateFit() {
+	if (props.fontSize !== 'auto' || props.text.length <= 0) return;
+	await document.fonts.ready;
+	adjustPadding();
+	adjustFontSize();
+
+	if (!resizeObserver) {
+		const container = labelContainer.value;
+		if (!container) return;
+
+		// Create a ResizeObserver to watch for changes in the container's dimensions
+		resizeObserver = new ResizeObserver(() => {
+			updateFit();
+		});
+
+		resizeObserver.observe(container);
+	}
+
+	adjustFontSize();
+}
+
+onMounted(() => {
+	updateFit();
+});
```

Comment:
- I would have guessed that we only need to wait for the fonts to be ready once on mount
- Let's create the observer on mount instead of checking on each `updateFit` call

```suggestion
function updateFit() {
	if (props.fontSize !== 'auto' || props.text.length <= 0) return;

	adjustPadding();
	adjustFontSize();
}

onMounted(async () => {
	const container = labelContainer.value;
	if (!container) return;

	await document.fonts.ready;

	// Create a ResizeObserver to watch for changes in the container's dimensions
	resizeObserver = new ResizeObserver(() => {
		updateFit();
	});

	resizeObserver.observe(container);
});
```

Reply:
We should only instantiate a resize observer for panels that need it, which is only panels with the font size set to auto. Your suggestion creates a resize observer that observes every label and metric panel regardless of their config. This would be a waste of browser resources. By including that logic in the updateFit function we can ensure that we check the necessary conditions every time it is mounted/updated and then only deploy a resize observer if needed. I have added additional logic to cleanup resize observers when if a panel no longer needs it but is still mounted.